canvas/gradient.go

```diff
@@ -62,99 +114,40 @@ func (g *Gradient) calculatePixel(w, h, x, y float64) *color.RGBA64 {
 	}
 
 	return pixel
-
-}
-
-func (g *Gradient) calculateGradient(x, y, w, h, ox, oy float64) float64 {
-	switch g.Direction {
-	case GradientDirectionVertical:
-		return linearVertical(x, y, w, h, ox, oy)
-	case GradientDirectionCircular:
-		return linearCircular(x, y, w, h, ox, oy)
-	default:
-		return linearHorizontal(x, y, w, h, ox, oy)
-	}
-}
-
-// pixelGradient is used during construction
-type pixelGradient struct {
-	g   *Gradient
-	img draw.Image // image cache during generator run
 }
 
-// NewLinearGradient returns a new Image instance that dynamically
-// renders a gradient of type GradientDirection
-func NewLinearGradient(start color.Color, end color.Color, direction GradientDirection) *Gradient {
-	pix := &pixelGradient{}
+func computeGradient(g Gradient, w, h int, startColor, endColor color.Color) image.Image {
```

Comment:
I think that here the startColor and endColor are superfluous as it's set in all Gradient types.
If Gradient is still an interface you could pass baseGradient, or if it's changed as above then the colours can be read directly